### ISSUE TITLE
Compilation breaks with OpenSSL 1.1

### ISSUE THREAD

**Issue by ge0rg** [dino/dino#10]
Hi,

when compiling the latest master, this happens:

```
/usr/src/dino/plugins/signal-protocol/src/signal_helper.c: In function ‘signal_vala_hmac_sha256_init’:
/usr/src/dino/plugins/signal-protocol/src/signal_helper.c:83:35: error: invalid application of ‘sizeof’ to incomplete type ‘HMAC_CTX {aka struct hmac_ctx_st}’
     HMAC_CTX *ctx = malloc(sizeof(HMAC_CTX));
                                   ^~~~~~~~
/usr/src/dino/plugins/signal-protocol/src/signal_helper.c:87:5: warning: implicit declaration of function ‘HMAC_CTX_init’ [-Wimplicit-function-declaration]
     HMAC_CTX_init(ctx);
     ^~~~~~~~~~~~~
```

This is probably a problem with using OpenSSL 1.1 despite not supporting the new API yet.

Possible fix: https://github.com/gdanezis/petlib/issues/13#issuecomment-275382040

**Comment by mar-v-in** [comment]
1a4cd25 fixes compilation with OpenSSL 1.1. Verified on ArchLinux + staging

**Comment by ge0rg** [comment]
Awesome, thanks. Builds on Debian as well now!
